Spoon Master/Controller/FavoriteViewController.swift

```diff
@@ -7,10 +7,118 @@
 //
 
 import UIKit
+import CoreData
 
 final class FavoriteViewController: UIViewController {
-
+    
+    @IBOutlet private weak var tableView: UITableView!
+    
+    private let context = (UIApplication.shared.delegate as? AppDelegate)?
+        .persistentContainer.viewContext
+    private var favRecipe = [FavoriteRecipe]()
+    private var repositories = Repositories(api: .share)
+    private var info: Recipe?
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        configView()
+        fetchContext()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        tableView.reloadData()
+        fetchContext()
+    }
+    
+    func configView() {
+        tableView.do {
+            $0.delegate = self
+            $0.dataSource = self
+            $0.register(UINib(nibName: Constant.Identifier.favoriteTableViewCell, bundle: nil),
+                        forCellReuseIdentifier: Constant.Identifier.favoriteTableViewCell)
+        }
+    }
+    
+    func fetchContext() {
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: Constant.CoreData.favoriteRecipe)
+        
+        do {
+            guard let newData = try context?.fetch(fetchRequest) as? [FavoriteRecipe] else { return }
+            favRecipe = newData
+        } catch {
+            print("Fetching context has error ")
+        }
+        tableView.reloadData()
+    }
+    
+    // MARK: - Fetch and config data
+    func configData(_ index: IndexPath) {
+        repositories.getRecipeInfo(recipeId: Int(favRecipe[index.row].recipeId)) { [weak self] (results) in
+            DispatchQueue.main.async {
+                self?.fetchData(results)
+            }
+        }
+    }
+    
+    func fetchData( _ result: BaseResult<Recipe>) {
+        switch result {
+        case .success(let value):
+            info = value
+        case .failure(let error):
+            print(error ?? "")
+        }
+        let storyboard = UIStoryboard(name: Constant.Storyboard.detail, bundle: nil)
+        if let detailVC = storyboard.instantiateViewController(
+            withIdentifier: Constant.Identifier.detailViewController) as?
+            DetailViewController, let newData = self.info {
+            
+            self.navigationController?.pushViewController(detailVC, animated: true)
+            detailVC.takeData(newData)
```

Comment:
```suggestion
            detailVC.takeData(newData)
            navigationController?.pushViewController(detailVC, animated: true)      
```
Set data rồi mới push chứ em?